### PR TITLE
Allow rejecting invites

### PR DESCRIPTION
**Pull request by illicitonion**
This is done by using the same /leave flow as you would use if you had
already accepted the invite and wanted to leave.

### CODE REVIEW

File: synapse/federation/federation_client.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def get_event_auth(self, destination, room_id, event_id):
         defer.returnValue(signed_auth)
 
     @defer.inlineCallbacks
-    def make_join(self, destinations, room_id, user_id, content):
+    def make_membership_event(self, destinations, room_id, user_id, membership, content):
```
Comment (erikjohnston) [inline synapse/federation/federation_client.py]: - Docstring?
- Can we assert that `membership in [Membership.Join, Membership.Leave]`?

File: synapse/federation/transport/client.py [hunk 2]
```diff
@@ -160,8 +160,8 @@ def make_query(self, destination, query_type, args, retry_on_dns_fail):
 
     @defer.inlineCallbacks
     @log_function
-    def make_join(self, destination, room_id, user_id, args={}):
-        path = PREFIX + "/make_join/%s/%s" % (room_id, user_id)
+    def make_membership_event(self, destination, room_id, user_id, membership, args={}):
+        path = PREFIX + "/make_%s/%s/%s" % (membership, room_id, user_id)
```
Comment (erikjohnston) [inline synapse/federation/transport/client.py]: Can we assert that `membership in [Membership.Join, Membership.Leave]`?

File: synapse/federation/federation_client.py (hunk 1 above)
Comment (illicitonion) [inline synapse/federation/federation_client.py]: Done

File: synapse/federation/transport/client.py (hunk 2 above)
Comment (illicitonion) [inline synapse/federation/transport/client.py]: Done

**Comment by erikjohnston** [comment]
LGTM

**Comment by illicitonion** [comment]
Manually merged

**Pull request closed without merging**